std/algorithm/searching.d

```diff
@@ -1792,27 +1799,44 @@ if (isRandomAccessRange!R1 && isBidirectionalRange!R2
     }
     // Stage 2: linear find
     size_t scout = needleLength - 1;
+outer:
     for (;;)
     {
         if (scout >= haystack.length)
-        {
-            return haystack[haystack.length .. haystack.length];
-        }
+            break;
         if (!binaryFun!pred(haystack[scout], needleBack))
         {
             ++scout;
             continue;
         }
         // Found a match with the last element in the needle
-        auto cand = haystack[scout + 1 - needleLength .. haystack.length];
-        if (startsWith!pred(cand, needle))
+        static if (isRandomAccessRange!R2)
         {
+            for (size_t j = scout + 1 - needleLength, k = 0; k < needleLength - 1; ++j, ++k)
```

Comment:
From experience it seems that loops with one induction variable and additions inside do better than loops with multiple induction variables. Could you please look into eliminating one of `j` and `k` and compute it on the fly inside by means of one addition?